packages/css/src/rules/types.ts

```diff
@@ -59,3 +59,133 @@ export type RulesVariants<RuleFn extends RuntimeFn<VariantGroups>> = Resolve<
 >;
 export type RecipeVariants<RecipeFn extends RuntimeFn<VariantGroups>> =
   RulesVariants<RecipeFn>;
+
+// == Tests ====================================================================
+if (import.meta.vitest) {
+  const { describe, it, assertType } = import.meta.vitest;
+
+  describe.concurrent("VariantSelection Type Test", () => {
+    type ExampleVariants = {
+      size: {
+        small: string;
+        large: string;
+      };
+      color: {
+        red: string;
+        blue: string;
+      };
+    };
+    function assertSelectedVariant(
+      variants: VariantSelection<ExampleVariants>
+    ) {
+      assertType<VariantSelection<ExampleVariants>>(variants);
+      return variants;
+    }
+
+    it("Valid VariantSelection Type", () => {
+      assertSelectedVariant({
+        size: "small",
+        color: "red"
+      });
+    });
+
+    it("Invalid VariantSelection Type", () => {
+      assertSelectedVariant({
+        // @ts-expect-error: selected variant value is not included in `size` key.
+        size: "medium", // error occured here
+        color: "red"
+      });
+    });
```

Comment:
> [!NOTE]
> **P3[Recommendations]: If you can, please do so. We encourage changes, but it's up to the developer to decide.**

Since we're specifying a type here rather than a generic, I don't think we need a function.
```typescript
    it("Valid VariantSelection Type", () => {
      assertType<VariantSelection<ExampleVariants>>({
        size: "small",
        color: "red"
      });
    });

    it("Invalid VariantSelection Type", () => {
      assertType<VariantSelection<ExampleVariants>>({
        // @ts-expect-error: selected variant value is not included in `size` key.
        size: "medium", // error occured here
        color: "red"
      });
    });
```

Reply:
Yeah I got what u mean